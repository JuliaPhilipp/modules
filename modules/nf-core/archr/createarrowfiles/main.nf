process ARCHR_CREATEARROWFILES {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "archr_dckr" // TODO update

    input:
    tuple val(meta), path(fragments)

    output:
    tuple val(meta), path("*.arrow")                , emit: arrowfile
    tuple val(meta), path("*.rds")                  , emit: arrowmetadata
    tuple val(meta), path("*.pdf")                   , emit: qualitycontrolplots
    tuple val(meta), path("*.R_sessionInfo.log")    , emit: session_info
    path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    template 'createarrowfiles.r'


    stub:
    """
    touch ${meta.id}.arrow
    touch ${meta.id}-Pre-Filter-Metadata.rds
    touch ${meta.id}-Fragment_Size_Distribution.pdf
    touch ${meta.id}-TSS_by_Unique_Frags.pdf
    touch ${meta.id}.R_sessionInfo.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ArchR: \$(Rscript -e "library(ArchR); cat(as.character(packageVersion('ArchR')))")
    END_VERSIONS
    """
}
