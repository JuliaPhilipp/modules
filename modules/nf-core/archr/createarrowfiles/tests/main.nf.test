// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test archr/createarrowfiles
nextflow_process {

    name "Test Process ARCHR_CREATEARROWFILES"
    script "../main.nf"
    process "ARCHR_CREATEARROWFILES"

    tag "modules"
    tag "modules_nfcore"
    tag "archr"
    tag "archr/createarrowfiles"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("createarrowfiles scATAC") {

        when {
            process {
                """

                input[0] = [
                    [ id:'test', genome:'hg19'], // meta map
                    file("/workspace/gitpod/nf-training/modules/modules/nf-core/archr/data/fragments.tsv.gz", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("cellrangeratac - scATAC fragments - stub") {

        options "-stub"

        when {
            process {
                """

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("/workspace/gitpod/nf-training/modules/modules/nf-core/archr/data/fragments.tsv.gz", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
