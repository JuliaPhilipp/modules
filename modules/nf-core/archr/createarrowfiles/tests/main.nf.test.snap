{
    "cellrangeratac - scATAC fragments - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "arrowfile": [
                    
                ],
                "arrowmetadata": [
                    
                ],
                "qualitycontrolplots": [
                    
                ],
                "session_info": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-20T13:47:46.043952312"
    },
    "createarrowfiles scATAC": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "arrowfile": [
                    
                ],
                "arrowmetadata": [
                    
                ],
                "qualitycontrolplots": [
                    
                ],
                "session_info": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-20T14:00:50.519933396"
    }
}